gitfile/gitfile.go

```diff
@@ -7,13 +7,15 @@ import (
 	"io"
 )
 
-// nolint:deadcode,unused
 func GetFileContents(ctx context.Context, repoUrl, filepath, ref string, callback func(url string)) (io.ReadCloser, error) {
```

Comment:
This being the main entry point into the library (I assume), we should comment the hell out of it. What's `ref`? What's `callback` used for? Why is there no `v ...interface{}` param here when the `Detect` has one?

Reply:
ok, will comment it